server/controllers/auth.js

```diff
@@ -96,6 +96,7 @@ exports.loginUser = asyncHandler(async (req, res, next) => {
 // @access Private
 exports.loadUser = asyncHandler(async (req, res, next) => {
   const user = await User.findById(req.user.id);
+  const userProfile = await Profile.findById(req.user.id);
```

Comment:
Would be good to validate the user id in the `req` object first before you go ahead and use it. If it's not supplied, then you return an error status to the client.

Reply:
Hi Sunday. This route uses a protect middleware to decode the token and send the req.user data.

Reply:
OK, @lipob. That means that validation has already taken place in the middleware. Thanks for letting me know this!

Reply:
You're welcome.